libraries/AP_ESC_Telem/AP_ESC_Telem.h

```diff
@@ -13,15 +18,64 @@ class AP_ESC_Telem {
 
     static AP_ESC_Telem *get_singleton();
 
+    // number of ESCs configured
+    uint8_t get_num_motors(void) const {
+        return _backends[_primary]->get_num_motors();
+    }
+
+    // get an individual ESC's temperature in degrees if available, returns true on success
+    bool get_temperature(uint8_t esc_index, int16_t& temp) const {
+        return _backends[_primary]->get_temperature(esc_index, temp);
+    }
+
+    // get an individual ESC's rpm if available, returns true on success
+    bool get_rpm(uint8_t esc_index, float& rpm) const {
+        return _backends[_primary]->get_rpm(esc_index, rpm);
+    }
+
+    // get an individual ESC's current in centi-amps if available, returns true on success
+    bool get_current_ca(uint8_t esc_index, uint16_t& amps_ca) const {
+        return _backends[_primary]->get_current_ca(esc_index, amps_ca);
+    }
+
     // get an individual ESC's usage time in seconds if available, returns true on success
-    bool get_usage_seconds(uint8_t esc_id, uint32_t& usage_sec) const;
+    bool get_usage_seconds(uint8_t esc_index, uint32_t& usage_sec) const {
+        return _backends[_primary]->get_usage_seconds(esc_index, usage_sec);
+    }
+
+    // return the average motor frequency in Hz for dynamic filtering
+    float get_average_motor_frequency_hz() const {
+        return _backends[_primary]->get_average_motor_frequency_hz();
+    }
+
+    // return all of the motor frequencies in Hz for dynamic filtering
+    uint8_t get_motor_frequencies_hz(uint8_t nfreqs, float* freqs) const {
+        return _backends[_primary]->get_motor_frequencies_hz(nfreqs, freqs);
+    }
+
+    // return the last time telemetry data was received in ms for the given ESC or 0 if never
+    uint32_t have_telem_data(uint8_t esc_id) const {
```

Comment:
You need to range check the `esc_id` on all of these. Also if this is returning a time, the function name should probably be more like `bool get_last_telem_update_ms(uint8_t esc_id, uint32_t &time_ms)`

Reply:
Fixed